py/client/pydeephaven/_arrow_flight_service.py

```diff
@@ -29,7 +30,9 @@ def import_table(self, data: pa.Table) -> Table:
             # No need to add headers/metadata here via the options argument;
             # or middleware is already doing it for every call.
             writer, reader = self._flight_client.do_put(
-                pa.flight.FlightDescriptor.for_path("export", str(ticket)), dh_schema)
+                pa.flight.FlightDescriptor.for_path("export", str(ticket)),
+                dh_schema,
+                FlightCallOptions(headers=self.session.grpc_metadata))
```

Comment:
Probably need to remove the comment above. Also I don't see it is wrapped for updating the metadata.

Reply:
Removed the comment.  You are right we were not updating the metadata for flight.  We are also not updating the metadata for non-flight calls (on the gRPC channel) in the existing code.   We can argue forever about this, I think the most consistent way to do this and the easiest way to get everybody to agree is to ensure we do it in ALL RPCs, both flight ("wrong" in this PR before the change I am about to make and as per the version of your comment) and non flight ("wrong" in the existing code).  So I am adding back the middleware to ensure that is done for everything, and we can approve and merge this review.

A counter argument goes like "But if we just refresh the token in a timer that should be enough".  I think that is worse than the current "if we do it in some calls it should be ok".  Again, my understanding of the auth protocol that the server is supposed to impose on the client is "the client should update it on every call".   So let's do it on every call.

Reply:
I am in favor of doing it for everything too. IMO, we now have 3 ways to keep a session alive:
1. the timer thread
2. plain gRPC calls
3. flight client calls (not the direct stream read/write, but doGet, doPut, doExchange)

I believe all 3 are needed to support most if not all session lifecycle management use-cases. So as long as we avoid the flight-authenticate-can't-be-called-more-than-once mine,  we have not only fixed the original issue but improved the Py client API.

Reply:
Cool, great, thanks, sounds like we are in agreement.